content/en/blog/2022/instrument-kafka-clients/index.md

```diff
@@ -0,0 +1,252 @@
+* using an external agent running alongside your application to add tracing.
+
+The former is mostly a "manual" approach which is about instrumenting your application directly.
+It means adding some specific dependencies to your project and make code changes.
+
+The latter is actually an "automatic" approach which is about not touching your application at all.
+The agent, running alongside the application, is able to intercept messages coming in and out and adds tracing information to them.
+
+## Instrumenting the Apache Kafka clients
+
+The OpenTelemetry project provides the `opentelemetry-kafka-clients-2.6` module which provides tracing instrumentation for Kafka clients.
+First, you need to add the corresponding dependency to your application.
+
+```xml
+<dependency>
+    <groupId>io.opentelemetry.instrumentation</groupId>
+    <artifactId>opentelemetry-kafka-clients-2.6</artifactId>
+</dependency>
+```
+
+Depending on the exporter that you want to use for exporting tracing information, you have to add the corresponding dependency as well.
+For example, in order to use the Jaeger exporter, the dependency is the following one.
+
+```xml
+<dependency>
+    <groupId>io.opentelemetry</groupId>
+    <artifactId>opentelemetry-exporter-jaeger</artifactId>
+</dependency>
+```
+
+In this way, you have the minimum set up for enabling the tracing on your Kafka based application.
+
+### Setting up the OpenTelemetry instance
+
+The entire tracing instrumentation is handled by an `OpenTelemetry` instance in your code.
+It needs to be created and registered globally in order to be made available to the Kafka clients instrumentation library.
+
+This can be done in two different ways:
+
+* using an SDK extension for environment-based auto-configuration.
+* using SDK builders classes for programmatic configuration.
+
+#### Using the SDK auto-configuration
+
+It is possible to configure a global `OpenTelemetry` instance by using environment variables thanks to the SDK extension for auto-configuration, enabled by adding the following dependency to your application.
+
+```xml
+<dependency>
+    <groupId>io.opentelemetry</groupId>
+    <artifactId>opentelemetry-sdk-extension-autoconfigure</artifactId>
+</dependency>
+```
+
+When the Kafka clients instrumentation library is used, it checks if there is an `OpenTelemetry` instance already created and registered.
+If it is not, the library code checks if the SDK auto-configure module is in the classpath and in this case initializes it for creating the `OpenTelemetry` instance automatically.
+The corresponding configuration happens through environment variables (or corresponding system properties).
+This is really a way to simplify the initialization of tracing. 
+
+The main environment variables to be set are the following:
+
+* `OTEL_SERVICE_NAME`: specify the logical service name. This is really useful when using a tracing UI (i.e. Jaeger UI) for showing data.
```

Comment:
> This is really useful when using a tracing UI (i.e. Jaeger UI) for showing data.

The service name is kind of mandatory, maybe you can phrase this even stronger?

Reply:
Actually in one of the previous discussion with other committers (I don't remember where and who :-D) they said that I should not make the point that service name is mandatory, which is kind of true, but at same time I agree with you. In general, it doesn't make sense not having a service name. For example, you could skip setting it but then in the Jaeger UI, for example, you have "unknown service" in the drop down list for selecting the service for getting the traces.

Reply:
yeah, "kind of mandatory", maybe write something like "It is recommended to set it"?